src/realm/geospatial.cpp

```diff
@@ -64,13 +64,18 @@ static bool type_is_valid(realm::StringData str_type)
 
 namespace realm {
 
+GeoPolygon GeoBox::to_polygon() const
+{
+    // We rely on the inversion rule here to ignore ordering of points.
+    // ie: A polygon that encompases more than a hemisphere is inverted.
+    return GeoPolygon{{{lo, GeoPoint{lo.longitude, hi.latitude}, hi, GeoPoint{hi.longitude, lo.latitude}, lo}}};
```

Comment:
I'm a bit unsure about this approach. 
Why not to just leave this type (GeoBox) as is in Geospatial, and only internally when needed for queries (for S2Region construction and for query for mongodb) use this conversion? 
The problem here was really that we used S2LatLng instead of S2Polygon, no? This change makes this api a bit inconsistent.